packages/template-retail-react-app/app/pages/checkout/index.test.js

```diff
@@ -511,63 +359,190 @@ test.skip('Can proceed through checkout as registered customer', async () => {
+                _type: 'orderAddress'
             }
-            currentBasket.shipments[0].shipping_address = shippingBillingAddress
-            currentBasket.billing_address = shippingBillingAddress
+            currentBasket.shipments[0].shippingAddress = shippingBillingAddress
+            currentBasket.billingAddress = shippingBillingAddress
+            return res(ctx.json(currentBasket))
+        }),
+
+        // mock add shipping method
+        rest.put('*/shipments/me/shipping-method', (req, res, ctx) => {
+            currentBasket.shipments[0].shippingMethod = defaultShippingMethod
+            return res(ctx.json(currentBasket))
+        }),
+
+        // mock get shipping methods
+        rest.get('*/shipments/me/shipping-methods', (req, res, ctx) => {
+            return res(ctx.delay(0), ctx.json(mockShippingMethods))
+        }),
+
+        // mock add payment instrument
+        rest.post('*/baskets/:basketId/payment-instruments', (req, res, ctx) => {
+            currentBasket.paymentInstruments = [
+                {
+                    amount: 0,
+                    paymentCard: {
+                        cardType: 'Master Card',
+                        creditCardExpired: false,
+                        expirationMonth: 1,
+                        expirationYear: 2030,
+                        holder: 'Test McTester',
+                        maskedNumber: '************5454',
+                        numberLastDigits: '5454',
+                        validFromMonth: 1,
+                        validFromYear: 2020
+                    },
+                    paymentInstrumentId: 'testcard1',
+                    paymentMethodId: 'CREDIT_CARD'
+                }
+            ]
             return res(ctx.json(currentBasket))
         }),
 
         // mock update address
         rest.patch('*/addresses/savedaddress1', (req, res, ctx) => {
             return res(ctx.json(mockedRegisteredCustomer.addresses[0]))
+        }),
+
+        // mock place order
+        rest.post('*/orders', (req, res, ctx) => {
+            currentBasket = {
+                ...scapiOrderResponse,
+                customerInfo: {...scapiOrderResponse.customerInfo, email: 'customer@test.com'}
+            }
+            return res(ctx.json(currentBasket))
+        }),
+
+        rest.get('*/baskets', (req, res, ctx) => {
+            const baskets = {
+                baskets: [currentBasket],
+                total: 1
+            }
+            return res(ctx.json(baskets))
         })
     )
 
     // Set the initial browser router path and render our component tree.
     window.history.pushState({}, 'Checkout', createPathWithDefaults('/checkout'))
     renderWithProviders(<WrappedCheckout history={history} />, {
-        wrapperProps: {siteAlias: 'uk', locale: {id: 'en-GB'}}
+        wrapperProps: {
+            // Not bypassing auth as usual, so we can test the guest-to-registered flow
+            bypassAuth: false,
```

Comment:
For a long time, I couldn't figure out why the checkout flow during tests was behaving strangely. It was as if I kept getting logged out (I had `isGuest: true` initially). 

It turns out that the default test behaviour is to bypass the auth flow and instead pass a token, which was unchanging, to the Commerce API provider. So here I disabled this behaviour by setting `bypassAuth` to false. Then I can test the flow of starting as a guest and then logging in during checkout.

Should we reconsider the default value of `bypassAuth`?

Reply:
That is the intention of bypassAuth, we don't want to go through the login flow. With checkout allow both guest and registered. It makes sense to turn the bypassAuth off and let the tests go through the normal flows

Reply:
I see. Then perhaps we need to `console.log` to inform developers that they're bypassing the login flow. Otherwise, they may not easily discover this behaviour.